DRILL-7169: Rename drill-root ArtifactID to apache-drill

vdiravka opened the pull request:
Change 'project.artifactId' from 'drill-root' to 'apache-drill'

_Note: it includes changes for #1743 also (it is expected that PR will be merged first)_

vdiravka commented:
> @vdiravka, with this change, a folder with the unpacked build in distribution was renamed from `apache-drill-1.16.0-SNAPSHOT` to `distribution-1.16.0-SNAPSHOT`, so for now built version is in the `distribution/target/distribution-1.16.0-SNAPSHOT/distribution-1.16.0-SNAPSHOT/` instead of `distribution/target/apache-drill-1.16.0-SNAPSHOT/apache-drill-1.16.0-SNAPSHOT/`.
> 
> I don't think that we should rename these folders.

Thanks for finding this, it is random mistake.

sohami commented on distribution/pom.xml:
```diff
@@ -390,71 +390,71 @@
                   <sources>
                     <source>
-                      <location>target/apache-drill-${project.version}/apache-drill-${project.version}/git.properties</location>
+                      <location>target/${project.parent.artifactId}-${project.version}/${project.parent.artifactId}-${project.version}/git.properties</location>
```
all the `<location>target/...` can be replaced by `<location>${project.build.directory}/....`

vdiravka replied:
Good suggestion. Done.

sohami commented on pom.xml:
```diff
@@ -42,7 +42,7 @@
 
-    <target.gen.source.path>${project.basedir}/target/generated-sources</target.gen.source.path>
+    <target.gen.source.path>${project.build.directory}/generated-sources</target.gen.source.path>
     <proto.cas.path>${project.basedir}/src/main/protobuf/</proto.cas.path>
```
is this path correct ? I don't see any `src/main/protobuf` path ?

Also there are other places like line [549](https://github.com/apache/drill/pull/1746/files#diff-600376dffeb79835ede4a0b285078036R549) which can be modified to replace `target/classes` with `${project.build.outputDirectory}`

vdiravka commented:
@vvysotskyi @sohami I have rebased the changes onto latest Drill master branch and added one commit with minor improvement.
Please review it, once the community will agreed in dev mailing list with these changes:
_Note:_ mail topic is `Rename 'project.artifactId'`

arina-ielchiieva commented:
@vdiravka / @vvysotskyi what is the status of this PR?

vvysotskyi commented:
Currently, there is no consensus for this change, here is mail thread with the discussion: https://lists.apache.org/thread.html/99a2098ae04d56dd5994288b5fc05780d7de09a2f4a5ffbc5ce39cde@%3Cdev.drill.apache.org%3E
